scripts/downsample_curriculum.py

```diff
@@ -0,0 +1,110 @@
+    parser.add_argument("--output-dir", type=Path, help='Outer directory to output downsampled curriculum to')
+    parser.add_argument("--num-samples", type=int, help='Number of samples to retrieve from input curriculum')
+    parser.add_argument("--random-seed", required=False, type=int, help='Optional random seed.')
+    parser.add_argument("--curriculum-name", required=False, type=str, help="Name of training curriculum")
+    parser.add_argument(
+        "-f", "--force",
+        action='store_true',
+        required=False,
+        help='Force overwrite of target directory. By default, the script exits with an error if there already exists '
+             'a directory at the target destination.'
+    )
+    args = parser.parse_args()
+
+    logging.basicConfig(level=logging.INFO)
+
+    input_dir: Path = args.input_dir
+    output_dir: Path = args.output_dir
+    num_samples: int = args.num_samples
+    random_seed: int = args.random_seed
+    force_overwrite: int = args.force
+    curriculum_name: str = args.curriculum_name
+    if not curriculum_name:
+        curriculum_name = str(output_dir).split('/')[-1]
+
+    if not isdir(str(input_dir)):
+        logging.warning("Input directory does not exist")
+        raise FileExistsError(str(input_dir))
+
+    if isdir(str(output_dir)):
+        if force_overwrite:
+            rmtree(str(output_dir))
+        else:
+            logging.warning("There already exists a directory in the target location")
+            raise FileExistsError(str(output_dir))
+
+    if num_samples < 0:
+        logging.warning("Attempting to retrieve a negative number of samples")
+        raise ValueError(num_samples)
+
+    if random_seed:
+        seed(random_seed)
+
+    object_to_target_directories: Mapping[str, Sequence[Path]] = dict()
+    selected_sample_count: int = 0
+    for feature_file_path in sorted(input_dir.glob("situation*")):
+        with open(feature_file_path / 'description.yaml', encoding="utf-8") as description_file:
+            description_yaml = yaml.safe_load(description_file)
+            target_object = description_yaml['language']
+            object_to_target_directories.setdefault(target_object, []).append(feature_file_path)
+
+    logging.info("Downsampling Objects")
+
+    for obj, potential_samples in sorted(object_to_target_directories.items(), key=lambda x: x[0]):
+        logging.info(f"Now processing {obj}")
+        if num_samples > len(potential_samples):
+            logging.warning(f"Attempting to retrieve more samples than exist for {obj}. Default to using all samples")
+            copy_all_directories(output_dir, potential_samples)
+            selected_sample_count += len(potential_samples)
+
+        else:
+            selected_samples: Sequence[Path] = sample(potential_samples, k=num_samples)
+            copy_all_directories(output_dir, selected_samples)
+            selected_sample_count += num_samples
```

Comment:
Since the two endling lines are virtually identical between the two and that's unlikely to change, I'd prefer to factor them out of the if/else and do `selected_samples = potential_samples` in the `num_samples > len(potential_samples)` case. Then you can just always do:
```python
copy_all_directories(output_dir, selected_samples)
selected_sample_count += len(selected_samples)
```
This makes the code a little easier to follow since it's explicit that both branches have this same logic.